### PR TITLE
[query] Refactor encoded block conversion to use options

### PR DESCRIPTION
**Pull request by arnikola**
Refactors the list of args taken in by encoded block conversion to avoid long argument lists and consolidate all option permutations into one object.

### CODE REVIEW

File: src/query/ts/m3db/types.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/ts/m3db/consolidators"
+)
+
+// Options describes the options for encoded block converters.
+type Options interface {
+	// SetLookbackDuration sets the lookback duration.
+	SetLookbackDuration(time.Duration) Options
+	// GetLookbackDuration returns the lookback duration.
+	GetLookbackDuration() time.Duration
```
Comment (benraskin92) [inline src/query/ts/m3db/types.go]: I would make these just `LookbackDuration()` (i.e. without the `Get` prefix) - I think that's Go standard convention

File: src/query/ts/m3db/options.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	opts.lookbackDuration = lookback
+	return &opts
+}
+
+func (o *encodedBlockOptions) GetLookbackDuration() time.Duration {
+	return o.lookbackDuration
+}
+
+func (o *encodedBlockOptions) SetConsolidationFunc(
+	fn consolidators.ConsolidationFunc,
```
Comment (benraskin92) [inline src/query/ts/m3db/options.go]: Can you put these on one line? We should come up with a convention..

Comment (benraskin92) [reply src/query/ts/m3db/options.go]: And the other places where you only have one argument.

File: src/query/ts/m3db/options.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	opts := *o
+	opts.pools = p
+	return &opts
+}
+
+func (o *encodedBlockOptions) GetIteratorPools() encoding.IteratorPools {
+	return o.pools
+}
+
+func (o *encodedBlockOptions) Validate() error {
```
Comment (benraskin92) [inline src/query/ts/m3db/options.go]: Check `LookbackDuration > 0`

Comment (arnikola) [reply src/query/ts/m3db/options.go]: Good call; I'll put the check on `SetLookbackDuration` and limit it to >= 0 instead

File: src/query/ts/m3db/types.go (hunk 1 above)
Comment (arnikola) [reply src/query/ts/m3db/types.go]: Good call, I swear I saw `Get..` in an existing `Options` so was going by that convention, but will update.

File: src/query/ts/m3db/options.go (hunk 2 above)
Comment (arnikola) [reply src/query/ts/m3db/options.go]: Was going by line length mostly, put it  on a single line now

Should get a standard convention at some point though

File: src/query/ts/m3db/encoded_block_iterator_builder.go [hunk 4]
```diff
@@ -59,10 +61,13 @@ func (b *encodedBlockBuilder) add(
 	str := start.String()
 	if _, found := b.blocksAtTime[str]; !found {
 		// Add a new encoded block
+		// NB: the lookback will always be 0, as splitting series into blocks is not
```
Comment (benraskin92) [inline src/query/ts/m3db/encoded_block_iterator_builder.go]: In the future, we plan to support this, right?

File: src/query/ts/m3db/options.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+		return false
+	}
+
+	return o.splitSeries
+}
+
+func (o *encodedBlockOptions) SetLookbackDuration(lookback time.Duration) Options {
+	// NB: lookback cannot be negative.
+	if lookback < 0 {
+		lookback = 0
```
[the remark this answers is not in the document]
Comment (arnikola) [reply src/query/ts/m3db/options.go]: Yeah fair enough, looks like that's how other Options do it

File: src/query/ts/m3db/encoded_block_iterator_builder.go (hunk 4 above)
Comment (arnikola) [reply src/query/ts/m3db/encoded_block_iterator_builder.go]: Potentially; it may not be worth dealing with the added complexity in this case.

**Pull request merged**